### PR TITLE
Story AT1-309: Analyse the result of two simultaneous observations by creating two subarrays, two CSP subarrays and two SDP subarrays. 

### PR DESCRIPTION
**Pull request by kalyanithigale**
1. Updated tmc-devices.json to add configuration for two CspSubarrayLeafNode and SdpSubarrayLeafNode.
2. Updated csp-devices.json to add configuration for two cspsubarrays and cbfsubarrays.( Referred configuration file from CSP repo.)
3. Updated sdp-devices.json to add configuration for two sdpsubarrays.(Referred configuration file from SDP repo.)
4. Updated csp-docker-compose.yml and sdp-docker-compose.yml to have SDP and CSP image versions which supports creation of two subarrays.
5. Updated cspsubarray device name to make it inline with CSP configuration file.

### CODE REVIEW

File: docker-compose/sdp-docker-compose.yml [hunk 1]
```diff
@@ -26,22 +26,41 @@ services:
...
-  sdpsubarray:
+
+  sdpsubarray1:
     image: ${DOCKER_REGISTRY_HOST}/sdp-prototype/tangods_sdp_subarray:0.5.16
     depends_on:
       - databaseds
       - rsyslog-tmcprototype
       - sdp-dsconfig
-    container_name: ${CONTAINER_NAME_PREFIX}sdp_subarray
+    container_name: ${CONTAINER_NAME_PREFIX}sdp_subarray1
```
Comment (mangeshska) [inline docker-compose/sdp-docker-compose.yml]: There is a need to come up with a standard for container naming, similar to device naming standards.

File: data/tmc-devices.json [hunk 2]
```diff
@@ -195,19 +194,19 @@
                         },
```
Comment (apurvapatkar) [inline data/tmc-devices.json]: "simulationMode" is not included in attribute_properties of Subarray 2. The issue is not exactly at this line. Need to add above property in attribute_properties block of Subarray 2.

File: data/tmc-devices.json [hunk 3]
```diff
@@ -195,19 +194,19 @@
...
                             ],
                             "CspSubarrayFQDN": [
-                                "mid_csp/elt/subarray01"
+                                "mid_csp/elt/subarray_02"
                             ],
                             "SdpSubarrayFQDN": [
-                                "mid_sdp/elt/subarray_1"
+                                "mid_sdp/elt/subarray_2"
                             ],
                             "SkaLevel": [
```
Comment (apurvapatkar) [inline data/tmc-devices.json]: SkaLevel should be 2 for SubarrayNode. Also, check the SkaLevel for other devices.

File: docker-compose/sdp-docker-compose.yml [hunk 4]
```diff
@@ -26,22 +26,40 @@ services:
...
+      - databaseds
+      - rsyslog-tmcprototype
+      - sdp-dsconfig
+    container_name: ${CONTAINER_NAME_PREFIX}sdp_subarray2
+    network_mode: ${NETWORK_MODE}
+    environment:
+    - TANGO_HOST=${TANGO_HOST}
+    - TOGGLE_CONFIG_DB=0
+    - TOGGLE_CBF_OUTPUT_LINK=0
+    - TOGGLE_AUTO_REGISTER=0
```
Comment (adityadangeska) [inline docker-compose/sdp-docker-compose.yml]: Maintain consistency in the indentation

File: docker-compose/sdp-docker-compose.yml [hunk 5]
```diff
@@ -26,22 +26,40 @@ services:
...
       - rsyslog-tmcprototype
       - sdp-dsconfig
-    container_name: ${CONTAINER_NAME_PREFIX}sdp_subarray
+    container_name: ${CONTAINER_NAME_PREFIX}sdp_subarray1
     network_mode: ${NETWORK_MODE}
     environment:
     - TANGO_HOST=${TANGO_HOST}
     - TOGGLE_CONFIG_DB=0
     - TOGGLE_CBF_OUTPUT_LINK=0
+    - TOGGLE_AUTO_REGISTER=0
```
Comment (adityadangeska) [inline docker-compose/sdp-docker-compose.yml]: Maintain consistency in the indentation

File: docker-compose/sdp-docker-compose.yml (hunk 4 above)
Comment (adityadangeska) [reply docker-compose/sdp-docker-compose.yml]: Check all files and maintain single indentation.

File: tmcprototype/CspSubarrayLeafNode/CspSubarrayLeafNode/CspSubarrayLeafNode.py [hunk 6]
```diff
@@ -209,6 +209,7 @@ def init_device(self):
         try:
             self._state = 0
             # create subarray Proxy
+            print ("self.CspSubarrayNodeFQDN :", self.CspSubarrayNodeFQDN)
```
Comment (adityadangeska) [inline tmcprototype/CspSubarrayLeafNode/CspSubarrayLeafNode/CspSubarrayLeafNode.py]: Remove unnecessary print

File: docker-compose/csplmc-docker-compose.yml [hunk 7]
```diff
@@ -22,8 +22,9 @@ services:
     volumes_from:
     - rsyslog-tmcprototype:rw
 
+
```
Comment (adityadangeska) [inline docker-compose/csplmc-docker-compose.yml]: Remove extra line break

File: docker-compose/sdp-docker-compose.yml [hunk 8]
```diff
@@ -26,6 +26,7 @@ services:
     entrypoint: ["python", "SDPMaster"]
     command: ["1", "-v4"]
 
+
```
Comment (adityadangeska) [inline docker-compose/sdp-docker-compose.yml]: Remove extra line break.

File: data/tmc-devices.json (hunk 2 above)
Comment (kalyanithigale) [reply data/tmc-devices.json]: 'simulationMode' property added in attribute_properties of subarray2.

File: data/tmc-devices.json (hunk 3 above)
Comment (kalyanithigale) [reply data/tmc-devices.json]: 'skalevel' updated for all the devices.

File: data/csp-devices.json [hunk 9]
```diff
@@ -52,43 +64,39 @@
...
-                                "mid_csp_pst/sub_elt/master"
-                            ],
                             "polled_attr": [
                                 "healthstate",
                                 "1000",
                                 "adminmode",
                                 "1000",
                                 "state",
-                                "1000",
+				"1000",
```
[the remark this answers is not in the document]
Comment (kalyanithigale) [reply data/csp-devices.json]: Corrected indentation.

File: docker-compose/csplmc-docker-compose.yml [hunk 10]
```diff
@@ -18,25 +18,43 @@ services:
       sh -c "wait-for-it.sh ${TANGO_HOST} --timeout=30 --strict --
              sudo ln -sf /var/run/rsyslog/dev/log /dev/log &&\
              tango_admin --ping-device mid_csp_cbf/sub_elt/master 10 &&\
-             /venv/bin/python /app/csplmc/CspMaster/CspMaster/CspMaster.py 01"
+             /venv/bin/python /app/csplmc/CspMaster/CspMaster/CspMaster.py csp"
```
[the remark this answers is not in the document]
Comment (kalyanithigale) [reply docker-compose/csplmc-docker-compose.yml]: We have referred devices.json from CSP repo which was suggested by Elisabetta. In there devices.json instance for CspMaster is given as 'csp'.

File: docker-compose/tmc-docker-compose.yml [hunk 11]
```diff
@@ -180,25 +180,44 @@ services:
...
     image: ${DOCKER_REGISTRY_HOST}/${DOCKER_REGISTRY_USER}/tmcprototype:latest
     depends_on:
-      - databaseds
-      - rsyslog-tmcprototype
-      - cspsubarray
-      - tmc-dsconfig
+    - databaseds
+    - rsyslog-tmcprototype
+    - cspsubarray1
+    - tmc-dsconfig
```
[the remark this answers is not in the document]
Comment (kalyanithigale) [reply docker-compose/tmc-docker-compose.yml]: Corrected indentation.

File: tmcprototype/CspSubarrayLeafNode/CspSubarrayLeafNode/CspSubarrayLeafNode.py [hunk 12]
```diff
@@ -593,7 +593,7 @@ def AssignResources(self, argin):
             #Parse receptorIDList from JSON string.
             jsonArgument = json.loads(argin[0])
             receptorIDList = jsonArgument[CONST.STR_DISH][CONST.STR_RECEPTORID_LIST]
-
+            print ("Assign Resources command called", receptorIDList)
```
[the remark this answers is not in the document]
Comment (kalyanithigale) [reply tmcprototype/CspSubarrayLeafNode/CspSubarrayLeafNode/CspSubarrayLeafNode.py]: Print statement removed.

File: tmcprototype/CspSubarrayLeafNode/CspSubarrayLeafNode.xmi [hunk 13]
```diff
@@ -31,7 +31,7 @@
     <deviceProperties name="CspSubarrayNodeFQDN" description="FQDN of the CSP Subarray Node Tango Device Server.">
       <type xsi:type="pogoDsl:StringType"/>
       <status abstract="false" inherited="false" concrete="true" concreteHere="true"/>
-      <DefaultPropValue>mid_csp/elt/subarray01</DefaultPropValue>
+      <DefaultPropValue>mid_csp/elt/subarray_01</DefaultPropValue>
```
[the remark this answers is not in the document]
Comment (kalyanithigale) [reply tmcprototype/CspSubarrayLeafNode/CspSubarrayLeafNode.xmi]: Default value removed

File: tmcprototype/SubarrayNode/SubarrayNode.xmi [hunk 14]
```diff
@@ -62,7 +62,7 @@
     <deviceProperties name="CspSubarrayNodeFQDN" description="This property stores the FQDN of CSP subarray Node device to which the Subarray Node is associated with.">
       <type xsi:type="pogoDsl:StringType"/>
       <status abstract="false" inherited="false" concrete="true" concreteHere="true"/>
-      <DefaultPropValue>mid_csp/elt/subarray01</DefaultPropValue>
+      <DefaultPropValue>mid_csp/elt/subarray_01</DefaultPropValue>
```
[the remark this answers is not in the document]
Comment (kalyanithigale) [reply tmcprototype/SubarrayNode/SubarrayNode.xmi]: default value removed.

File: docker-compose/sdp-docker-compose.yml (hunk 4 above)
Comment (kalyanithigale) [reply docker-compose/sdp-docker-compose.yml]: Corrected indentation.

File: docker-compose/sdp-docker-compose.yml (hunk 5 above)
Comment (kalyanithigale) [reply docker-compose/sdp-docker-compose.yml]: Corrected indentation.

File: tmcprototype/CspSubarrayLeafNode/CspSubarrayLeafNode/CspSubarrayLeafNode.py (hunk 6 above)
Comment (kalyanithigale) [reply tmcprototype/CspSubarrayLeafNode/CspSubarrayLeafNode/CspSubarrayLeafNode.py]: Print statement removed.

File: docker-compose/csplmc-docker-compose.yml (hunk 7 above)
Comment (kalyanithigale) [reply docker-compose/csplmc-docker-compose.yml]: Removed.

File: docker-compose/sdp-docker-compose.yml (hunk 8 above)
Comment (kalyanithigale) [reply docker-compose/sdp-docker-compose.yml]: Removed.

**Comment by kalyanithigale** [comment]
All the comments are resolved in commit id: https://github.com/ska-telescope/tmc-prototype/commit/5b0f070a4ecd6fc08f2330c7301c1a4b926eac3c

**Comment by apurvapatkar** [comment]
Looks ok to me.

**Pull request merged**